### PR TITLE
Intelligently splits flac file such that no individual file exceeds 100mb

### PR DESCRIPTION
**Pull request by yoninachmany**
r @sandlerben @roornelas

IBM only accepts files that are 100mb in size -To catch all the words, repeats a 5 s section. -SplitFlacFile converts file to Wav, gets number of chunks including redundant 5 seconds using a helper function, and extracts audio segments of a given length using a derivation (see comment) and another helper function
in process of mock testing!

### CODE REVIEW

**Comment by sandlerben** [comment]
@yoninachmany look at the diff and explain how it makes any sense? It seems like this branch is making changes to config only (changes which are no longer necessary since I made a `config` package).

**Comment by yoninachmany** [comment]
@sandlerben not completely sure what happened, code was pushed in this commit (https://github.com/hack4impact/transcribe4all/pull/51/commits/75b02b2506fba48b60f643bb49f94cdbe8265fa0), perhaps an issue came up when `transcription.go` was renamed to 'utils.go'

Anyway, I copied code from that commit into `utils.go`

**Comment by sandlerben** [comment]
@yoninachmany okay here is what I want you to do because the diff is still not right.
1. `git checkout split-file`
2. `git fetch && git reset --hard origin/master`
3. `git cherry-pick ef60c51ffac32df28610de9ef57e2ae7d0a28e17`
4. `git push -f origin HEAD`

**Comment by sandlerben** [comment]
@yoninachmany looks pretty good. Have you tested this locally?

File: transcription/utils.go [hunk 1]
```diff
@@ -86,6 +86,73 @@ func filePathFromURL(url string) string {
...
+	// http://stackoverflow.com/questions/36632511/split-audio-file-into-several-files-each-below-a-size-threshold
+	// The Stack Overflow answer ultimately calculated the length of each audio chunk in seconds.
+	// chunk_length_in_sec = math.ceil((duration_in_sec * file_split_size ) / wav_file_size)
+	// Invariant: If ConvertAudioIntoWavFormat is called on fn, a 95MB chunk of resulting Wav file is always 2968 seconds.
+	// In the above equation, there is one constant: file_split_size = 95000000 bytes.
+	// duration_in_sec is used to calculate wav_file_size, so it is canceled out in the ratio.
+	// wav_file_size = (sample_rate * bit_rate * channel_count * duration_in_sec) / 8
+	// sample_rate = 44100, bit_rate = 16, channels_count = 1 (stereo: 2, but Sphinx prefers 1)
+	// As a chunk of the Wav file is extracted using FFMPEG, it is converted back into Flac format.
+	wavFileName, err := ConvertAudioIntoFormat(fn, "wav")
```
Comment (sandlerben) [inline transcription/utils.go]: let's actually let the caller do the conversions to and from wav

speaking of, can you make sure this function is called where it needs to be called?

File: transcription/utils.go [hunk 2]
```diff
@@ -86,6 +86,73 @@ func filePathFromURL(url string) string {
...
+	if err != nil {
+		return err
+	}
+
+	chunkLengthInSeconds := 2968
+	for i := 0; i < numChunks; i++ {
+		// 5 seconds of redundancy for each chunk after the first
+		startingSecond := i*chunkLengthInSeconds - (i-1)*5
+		newFileName := wavFileName + strconv.Itoa(i)
+		if err := extractAudioSegment(newFileName, startingSecond, chunkLengthInSeconds); err != nil {
```
Comment (sandlerben) [inline transcription/utils.go]: how will the caller know what the produced files are? should we return a slice of strings indicating the file paths in order?

File: transcription/utils.go [hunk 3]
```diff
@@ -86,6 +86,73 @@ func filePathFromURL(url string) string {
...
+		}
+		if _, err := ConvertAudioIntoFormat(newFileName, "flac"); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// getNumChunks gets file size in MB, divides by 95 MB, and add 1 more chunk in case
+func getNumChunks(fn string) (int, error) {
```
Comment (sandlerben) [inline transcription/utils.go]: fn -> filePath to be consistent with other functions

**Comment by yoninachmany** [comment]
@sandlerben 
re "make sure this function is called where it needs to be called":
how will this section (https://github.com/hack4impact/transcribe4all/blob/master/transcription/utils.go#L110-L128) need to be adjusted for a slice of `flacPaths` rather than a single `flacPath`?

**Comment by sandlerben** [comment]
basically run `TranscribeWithIBM` in a loop for each path

**Pull request merged**